src/compiler.jl

```diff
@@ -4746,15 +4746,15 @@ end
         else
             return quote
                 Base.@_inline_meta
-                sret = Box{$(Tuple{sret_types...})}()
+                sret = Box{$(AnonymousStruct(Tuple{sret_types...}))}()
```

Comment:
Can we just have this subsume the prior special casing of if it contains an any or not allocated inline as well?

Reply:
Probably, want to make that change?